python/lsst/analysis/tools/actions/keyedData/calcCompletenessHistogram.py

```diff
@@ -0,0 +1,127 @@
+# This file is part of analysis_tools.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = ("CalcCompletenessHistogramAction", "MagnitudeCompletenessConfig")
+
+import copy
+
+import lsst.pex.config as pexConfig
+import numpy as np
+from lsst.pex.config.configurableActions import ConfigurableActionField
+
+from ...interfaces import KeyedData, KeyedDataAction, KeyedDataSchema
+from ...math import isPercent
+from ..config import MagnitudeBinnedMetricsConfig
+from .calcBinnedCompleteness import CalcBinnedCompletenessAction
+
+
+class MagnitudeCompletenessConfig(pexConfig.Config):
+    """Configuration for measuring magnitudes at given completeness
+    thresholds."""
+
+    completeness_mag_max = pexConfig.Field[float](
+        doc="Brightest magnitude to consider checking if completeness is below a percentile threshold for",
+        default=18,
+    )
+    completeness_percentiles = pexConfig.ListField[float](
+        doc="The percentiles to find the magnitude at.",
+        default=[90.0, 80.0, 50.0, 20.0, 10.0],
+        itemCheck=isPercent,
+    )
+
+
+class CalcCompletenessHistogramAction(KeyedDataAction):
+    """Action to calculate a histogram of completeness vs magnitude."""
+
+    action = ConfigurableActionField[CalcBinnedCompletenessAction](
+        doc="The action to compute completeness/purity",
+    )
+    bins = pexConfig.ConfigField[MagnitudeBinnedMetricsConfig](
+        doc="The bin configuration",
+    )
+    config_metrics = pexConfig.ConfigField[MagnitudeCompletenessConfig](doc="Metric definition configuration")
+
+    def __call__(self, data: KeyedData, **kwargs) -> KeyedData:
+        band = kwargs.get("band")
+        bins = tuple(x / 1000.0 for x in reversed(self.bins.get_bins()))
```

Comment:
Should the number of bins be a config option?

Reply:
The bins are returned by a config class.